src/components/Navigation/Navigation.test.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+
+import Navigation from './Navigation.component';
+
+test('check the drawer status after click in drawer openner', async () => {
+  render(<Navigation />);
+  const drawerOpener = await screen.getByRole('drawer-opener');
+  fireEvent.click(drawerOpener);
+  /**
+   * check drawer after drawer opener is clicked, drawe must be rendered,
+   * if drawer is declared before the click event the test wil be false
+   */
+  const drawer = await screen.getAllByRole('drawer-element');
+  expect(drawer).toBeTruthy();
```

Comment:
A nice way to check if this is really what's happening, is to validate that the component wasn't present before the click. You can wrap the `fireEvent.click` in an `act` (which comes with the react-testing-library as well), and `queryByRole` for that element. It will return `null` if the element doesn't exist instead of throwing an error as `getByRole` does. Then, after the `fireEvent.click` you do `getByRole` and expect that the element is present.